### PR TITLE
Update advanced-usage.rst

### PR DESCRIPTION
**Pull request by janbbeck**
Just a few changes to address run-on sentences, punctuation and spelling.

Unresolved issues. If this is too pedantic, just let me know...

This document mixes whether 'SSH-MITM proxy server' is a name by itself or not. For example, consider this sentence:

"SSH-MITM proxy server is capable of advanced man-in-the-middle attacks."
This is fine, treating 'SSH-MITM proxy server' as a name or term-of-the-art.
Consider this sentence:
"If agent forwarding is not possible, the SSH-MITM proxy server can accept the
public key authentication request and redirect the session to a honey pot."
Also fine, but now its descriptive rather than a name.
Finally, in some places we have just SSH-MITM as a name, which of course it is. But since it's 'SSH-MITM proxy server' everywhere else, we should probably add proxy server here or remove it almost everywhere else....

My suggestion would be to describe it as 'The SSH-MITM proxy server' in the first sentence, and then just call it 'SSH-MITM' from there on.

Again, possibly too pedantic....

### CODE REVIEW

File: doc/advanced-usage.rst [hunk 1]
```diff
@@ -78,7 +78,7 @@ when the integrity of a machine is not trusted. (https://tools.ietf.org/html/dra
     of logging and access control.  This access control may utilize
     information about the path through which the connection was received (as
     collected with SSH_AGENT_FORWARDING_NOTICE messages; however, the path
-    is reliable only up to and including the first unreliable machine.).
+    is reliable only up to and including the first unreliable machine).
```
Comment (manfred-kaiser) [inline doc/advanced-usage.rst:81]: This is a quote from https://datatracker.ietf.org/doc/html/draft-ietf-secsh-agent-02
I think we should not change quoted text.

Comment (janbbeck) [reply doc/advanced-usage.rst:81]: Fair enough. Let me change that.

**Comment by manfred-kaiser** [comment]
Thanks for spell checking :smiley: 

 I will wait merging your pull request and merge all changes at one, if this is ok for you.

**Comment by janbbeck** [comment]
> Thanks for spell checking smiley
> 
> I will wait merging your pull request and merge all changes at once, if this is ok for you.

You mean all files?

**Comment by manfred-kaiser** [comment]
> > Thanks for spell checking smiley
> > I will wait merging your pull request and merge all changes at once, if this is ok for you.
> 
> You mean all files?

:smile: I mean, I will wait until you write that the pull request is ready to merge

**Comment by janbbeck** [comment]
Ah, ok. Well I did make that change, but we still have to decide about the naming issues that I described in the pull request comment. Once we make a decision there I will finish the changes and let you know. OK?

**Comment by manfred-kaiser** [comment]
I think we should remove **SSH-MITM proxy server** and replace it with **SSH-MITM**

If someone is reading 'proxy server' this implies that it can be used as a proxy for ssh connections (jump host).

SSH-MITM should not be used as a proxy/jump host.

**Comment by janbbeck** [comment]
Should be ok now.

**Pull request merged**